core/src/main/java/io/confluent/kafka/schemaregistry/metrics/MetricsContainer.java

```diff
@@ -153,15 +147,8 @@ public SchemaRegistryMetric getNodeCountMetric() {
     return nodeCount;
   }
 
-  public void setLeader(boolean leader) {
-    isLeaderNode.set(leader ? 1 : 0);
-    if (telemetryReporter != null) {
-      if (leader) {
-        telemetryReporter.contextChange(metricsContext);
-      } else {
-        telemetryReporter.close();
-      }
-    }
+  public SchemaRegistryMetric getLeaderNode() {
+    return isLeaderNode;
```

Comment:
nit, any reason we don't want to keep the existing setLeader signature and set `isLeaderNode.set(leader ? 1 : 0);` as part of this method?

Reply:
I modified it to be similar to all other metric getters in the file. This one was special because it had that leader change logic, but now it's not anymore.